homeassistant/components/unifiprotect/models.py

```diff
@@ -10,3 +21,31 @@ class ProtectRequiredKeysMixin:
 
     ufp_required_field: str | None = None
     ufp_value: str | None = None
+    ufp_value_callable: Callable[[ProtectAdoptableDeviceModel | NVR], Any] | None = None
+
+    def get_ufp_value(self, obj: ProtectAdoptableDeviceModel | NVR) -> Any:
+        """Return value from UniFi Protect device."""
+        if self.ufp_value is not None:
+            return get_nested_attr(obj, self.ufp_value)
+        if self.ufp_value_callable is not None:
+            return self.ufp_value_callable(obj)
+        return None  # pragma: no cover
```

Comment:
Can we avoid the no cover ?

Reply:
Not really testable. Even if I made it raise an exception, `ufp_value` and `ufp_value_callable` can only both be None someone configs it that way.

Reply:
I probably should make it raise an exception though. So, future us knows one of them is required in case we forget.